internal/web/cacheverifier.go

```diff
@@ -0,0 +1,210 @@
+	tickC := tick.C
+	defer tick.Stop()
+
+	var diffErr *cacheVerificationError
+
+endlessLoop:
+	for range tickC {
+		if nonCachedItems, err := verifier.listItems(ctx, nonCachedClient); err != nil {
+			fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed to get actual items: %v\n", err)
+		} else if cachedItems, err := verifier.listItems(ctx, cachedClient); err != nil {
+			fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed get cached items: %v\n", err)
+		} else {
+		outerLoop:
+			for _, nonCachedItem := range nonCachedItems {
+				for _, cachedItem := range cachedItems {
+					if nonCachedItem.GetNamespace() == cachedItem.GetNamespace() &&
+						nonCachedItem.GetName() == cachedItem.GetName() {
+						if nonCachedItem.GetResourceVersion() != cachedItem.GetResourceVersion() {
+							fmt.Fprintf(os.Stderr, "CACHEVERIFIER: resource versions of %s differ, need to check\n",
+								cachedItem.GetName())
+							if err := verifier.handleDiff(cachedItem, nonCachedItem); errors.As(err, &diffErr) {
+								break endlessLoop
+							} else if err != nil {
+								fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed to handle diffs: %v\n", err)
+							}
+						} else {
+							// previously stored inconsistency of this item can now be deleted
+							verifier.diffMap.Delete(cache.MetaObjectToName(cachedItem).String())
+						}
+						continue outerLoop
+					}
+				}
+				fmt.Fprintf(os.Stderr, "CACHEVERIFIER: non-cached item %s not found in cache, need to check\n",
+					nonCachedItem.GetName())
+				// non cached package found, that is not in cache (yet?)
+				if err := verifier.handleDiff(nil, nonCachedItem); errors.As(err, &diffErr) {
+					break endlessLoop
+				} else if err != nil {
+					fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed to handle diffs: %v\n", err)
+				}
+			}
+
+			// check the other way around (i.e. whether there are cached items, that are not in the non-cached list anymore)
+			for _, cachedClPkg := range cachedItems {
+				found := false
+				for _, nonCachedClPkg := range nonCachedItems {
+					if nonCachedClPkg.GetNamespace() == cachedClPkg.GetNamespace() &&
+						nonCachedClPkg.GetName() == cachedClPkg.GetName() {
+						found = true
+						break
+					}
+				}
+				if !found {
+					fmt.Fprintf(os.Stderr, "CACHEVERIFIER: cached item %s not found in non-cached list, need to check\n",
+						cachedClPkg.GetName())
+					// when here, there exists a cached item, that is not in the non-cached list
+					if err := verifier.handleDiff(cachedClPkg, nil); errors.As(err, &diffErr) {
+						break endlessLoop
+					} else if err != nil {
+						fmt.Fprintf(os.Stderr, "CACHEVERIFIER: failed to handle diffs: %v\n", err)
+					}
+				}
+			}
+		}
+	}
+	verifier.closeAndReset(diffErr)
+}
+
+func (verifier *cacheVerifier) closeAndReset(err error) {
+	close(verifier.stopCh)
+	verifier.diffMap.Range(func(key, value any) bool {
+		verifier.diffMap.Delete(key)
+		return true
+	})
```

Comment:
alternatively you could just reset to a new empty `sync.Map{}` let the GC do the rest